### ISSUE TITLE
Pulling Request Comment via GitHub REST API

### ISSUE THREAD

**Issue by RavenMarQ** [sailuh/kaiaulu#323]
Placeholder

**Comment by RavenMarQ** [comment]
Quick questions on the specification of this issue:
> Is this similar to parsing using SciTools' Understand to create data tables? 
> Looking through the API, there are four data formats:
    - Raw Markdown
    - text representation of raw markdown
    - HTML render of raw markdown
    - Full json = all three data sets above (larger)
Are there any specific ones I need focus on or should I just use the json?
> I can make the request for the data via cURL in command line (similar to what I have done for Understand), JavaScript and [Octokit](https://github.com/octokit/), or GitHub CLI. Which of these should I focus on? Or should I do research on all of them?
> With the API, I can GET a review comment and also GET a list. Should I be first retrieving a list then getting the specified review comments from the list?

**Comment by carlosparadis** [comment]
No, this is an API request to download data. It is closer to @daomcgill if anything, but she is not using API. You should look at the pull request for GitHub. It has many examples on how to write R code to download data. Please follow the format already used on github.R (i'd recommend trying the notebook from a user perspective too and downloading some data). Again see the PR, the GitHub code has not been merged yet). @crepesAlot  pinging you here so you are aware too.

**Comment by RavenMarQ** [comment]
Taking a glance at the [github.R PR](https://github.com/sailuh/kaiaulu/pull/301/files#diff-0e7f263cca4715298605bdeb71c7f58601660c9cb23607acd7b47f820cd8ae9e), it's not well-documented but I'll report back what I can scrounge up of what I understand. If this is not the file you were meaning for me to look at, let me know.

**Comment by carlosparadis** [comment]
not exec/github.R, look at R/github.R and the associated notebook. @beydlern should have revised it.

**Comment by carlosparadis** [comment]
also please collaborate closely with @crepesAlot once either of you understand how it has been done, explaining to the other will save a lot of your time.

**Comment by beydlern** [comment]
@RavenMarQ 
The notebooks that I refactored are located in PR #289.
- vignettes/download_github_comments.Rmd
- vignettes/gitlog_entity_showcase.Rmd
- vignettes/gitlog_showcase.Rmd

**Comment by carlosparadis** [comment]
Look at vignettes/download_github_comments.Rmd. The effort in this task should be very minimal and non code related, since you are not writing the GitHub API interface from scratch like @beydlern  is doing with OpenHub. There is already a lot of code written to interact with GitHub API. Just see how the API endpoint looks like in R/github.R for one endpoint, and literally copy and paste the code and change the url and adapt the documentation. 

Then you need to write the parser for the downloaded json, which is something you're familiar with from having to do the XML. In hindsight, @beydlern is the closest task equivalent to yours, since he is also using an API. I'd hope the R/github.R code suffices to give a template on how to do that.

**Comment by carlosparadis** [comment]
@crepesAlot when you can please ask Raven to update this issue specification. For now i'd recommend at least keep track on a comment here, and I can hyperlink the first comment until the final version is done.

**Comment by crepesAlot** [comment]
@carlosparadis Here's the main points that are retrieved from this endpoint.
Using the following Pull Request: https://github.com/sailuh/kaiaulu/pull/327/files/144d076fe0d1c9096d4a8091c45d9ec7adfa2c52 
![Image](https://github.com/user-attachments/assets/d5883cea-27ed-4600-81e9-b629c832c03f)
The three columns are retrieved as such:
![Image](https://github.com/user-attachments/assets/4a7d3741-47f8-4722-90cb-2223a753a4f3)

**Comment by crepesAlot** [comment]
@carlosparadis I have tried testing what you asked about how the information for the `diff_hunk` field changes when clicking a single line and selecting multiple lines.
Somehow testing this on my own repo has only further confused me.
In this screenshot, I selected lines 22 to 30 when making the comment.
![Image](https://github.com/user-attachments/assets/dfdcff49-6229-4c3a-a1fb-440c60d03bcc)
Yet the response in the `diff_hunk` field return every line in the entire file until line 30. It responded like this regardless of whether I selected multiple lines or clicked once.
![Image](https://github.com/user-attachments/assets/31555bf8-b680-4f2b-9f2f-74c076641416)
I'm not sure why this occurs, as it doesn't do this for any of the responses from Kaiaulu

**Comment by carlosparadis** [comment]
Ah yes, another GitHub mystery! 

https://github.com/sailuh/kaiaulu/issues/323#issuecomment-2513401909 

Why here my comment appear twice in the table?

And https://github.com/sailuh/kaiaulu/issues/323#issuecomment-2513418793 

What is the docs definition for this diff hunk?

**Comment by carlosparadis** [comment]
@crepesAlot remind me again, was this ready for review or was there anything pending?

**Comment by crepesAlot** [comment]
@carlosparadis I just need to update the function documentation for more in depth explanation and details on the different fields.

**Comment by crepesAlot** [comment]
@carlosparadis I updated the function documentations and NEWS.md, the code review should be good to go!

**Comment by crepesAlot** [comment]
@carlosparadis I was able to find what connects the Pull Request Comment endpoint (this endpoint) with the Pull Request Review endpoint.
![Image](https://github.com/user-attachments/assets/1ec41481-2acb-4da2-8108-6d20afdcee98)
![Image](https://github.com/user-attachments/assets/4f3daa12-6ac1-45fd-8fbe-d937c2cff943)
I've updated the Pull Request Comments parser to add the `pull_request_review_id` field as `review_id` in the table.
